build.rs

```diff
@@ -1,6 +1,19 @@
+use regex::Regex;
 use rustc_version::version;
-use std::env;
-const RUSTUP_TOOLCHAIN_ID: &str = "RUSTUP_TOOLCHAIN";
+use std::process::Command;
+
+fn detect_aurix_toolchain() -> Result<Option<String>, ()> {
+    Command::new("rustup")
+        .args(["toolchain", "list"])
+        .output()
+        .map_or(Err(()), |result| {
+            let re = Regex::new(r"tricore-htc-none.+").unwrap();
+            let result =
+                String::from_utf8(result.stdout).expect("Unable to convert to utf8 string");
```

Comment:
```suggestion
                String::from_utf8_lossy(&result.stdout);
```

Reply:
I will not accept this because it can silently create a corrupted string that will not trigger any test. I really like to have crash because it is a sign of something bad.